redisgears_core/src/lib.rs

```diff
@@ -938,21 +943,36 @@ fn js_init(ctx: &Context, _args: &[RedisString]) -> Status {
         Err(_) => ctx.log_notice("Failed loading RedisAI API."),
     }
 
+    let version = match get_and_verify_redis_version_compatible(ctx) {
+        Ok(v) => v,
+        Err(e) => {
+            log::error!("{e}");
+            return Status::Err;
+        }
+    };
+
+    let is_enterprise = ctx
+        .server_info("server")
+        .field("rlec_version")
+        .map_or(false, |_| true);
+
+    let redis_version = RedisVersion {
+        version,
+        is_enterprise,
+    };
+
```

Comment:
Perhaps, we could inline this code directly into the `get_and_verify_redis_version_compatible` function? And make this function return `RedisVersion` instead of the `Version`?